From fc477991d74b55cc96fdbcde7aefc4efad619d4e Mon Sep 17 00:00:00 2001
From: xjljian <xjlxxx3@gmail.com>
Date: Sun, 24 Nov 2013 13:34:29 +0800
Subject: [PATCH] hwcomposer: Add sourceTransform to hwc_layer_t

---
 include/hardware/hwcomposer_v0.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/hardware/hwcomposer_v0.h b/include/hardware/hwcomposer_v0.h
index 473819b..75e057c 100644
--- a/include/hardware/hwcomposer_v0.h
+++ b/include/hardware/hwcomposer_v0.h
@@ -100,6 +100,11 @@ typedef struct hwc_layer {
             /* transformation to apply to the buffer during composition */
             uint32_t transform;
 
+#ifdef QCOM_HARDWARE
+            /* source transform of the buffer */
+            uint32_t sourceTransform;
+#endif
+
             /* blending to apply during composition */
             int32_t blending;
 
-- 
1.8.1.2

