From 71103e10ffaa205527f6eb0ec11d10a31b76bf1b Mon Sep 17 00:00:00 2001
From: xjljian <xjlxxx3@gmail.com>
Date: Sat, 11 Jan 2014 00:22:59 +0800
Subject: [PATCH 1/2] support legacy qcom audio variant

---
 core/qcom_target.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/qcom_target.mk b/core/qcom_target.mk
index e65a6fc..08680b0 100644
--- a/core/qcom_target.mk
+++ b/core/qcom_target.mk
@@ -12,7 +12,7 @@ ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
     endif
 
     # Enable DirectTrack for legacy targets
-    ifneq ($(filter caf bfam,$(TARGET_QCOM_AUDIO_VARIANT)),)
+    ifneq ($(filter caf bfam legacy,$(TARGET_QCOM_AUDIO_VARIANT)),)
         ifeq ($(BOARD_USES_LEGACY_ALSA_AUDIO),true)
             TARGET_GLOBAL_CFLAGS += -DQCOM_DIRECTTRACK
             TARGET_GLOBAL_CPPFLAGS += -DQCOM_DIRECTTRACK
-- 
1.8.1.2

