From 6eb674a055f5e1d3a7018e8cb79ee6748f2f73b7 Mon Sep 17 00:00:00 2001
From: xjljian <xjlxxx3@gmail.com>
Date: Fri, 20 Dec 2013 22:03:54 +0800
Subject: [PATCH] wifi: add support for custom Wi-Fi implementations

---
 wifi/Android.mk | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/wifi/Android.mk b/wifi/Android.mk
index c7bdc59..7a03b88 100644
--- a/wifi/Android.mk
+++ b/wifi/Android.mk
@@ -40,7 +40,13 @@ ifdef WIFI_EXT_MODULE_NAME
 LOCAL_CFLAGS += -DWIFI_EXT_MODULE_NAME=\"$(WIFI_EXT_MODULE_NAME)\"
 endif
 
-LOCAL_SRC_FILES += wifi/wifi.c
+ifeq ($(TARGET_CUSTOM_WIFI),)
+       LOCAL_SRC_FILES += \
+               wifi/wifi.c
+else
+       LOCAL_SRC_FILES += \
+               $(TARGET_CUSTOM_WIFI)
+endif
 
 ifeq ($(BOARD_HAVE_SAMSUNG_WIFI),true)
 LOCAL_CFLAGS += -DSAMSUNG_WIFI
-- 
1.8.1.2

